packages/fxa-auth-server/lib/routes/attached-clients.js

```diff
@@ -93,7 +99,21 @@ module.exports = (log, db, devices, clientUtils) => {
             clientUtils.formatLocation(...args);
           },
           deviceList: async () => {
-            return await request.app.devices;
+            let devices = await request.app.devices;
+            
+            // To help reduce duplicate devices and help improve send tab 
+            // performance a client can request to filter device last access 
+            // time by a specified number of days. For reference, Sync currently
+            // considers devices that have been accessed in the last 21 days to 
+            // be active.
+            const idleDeviceTimestamp = request.query.filterIdleDevicesTimestamp;
+            if (idleDeviceTimestamp) {
+              devices = devices.filter((device) => {
+                return device.lastAccessTime > idleDeviceTimestamp;
```

Comment:
Would it be better to add this logic to the sproc? Even though a 500 device limit has been created, it seems like a lot of devices could still be needlessly queried and returned from the DB.

Reply:
Actually, looking at the code a bit more, the devices returned from `request.app.devices` is a combination of [devices and session tokens](https://github.com/mozilla/fxa/blob/d693ce671ec67ce00730cce13a108a4067fdfa94/packages/fxa-auth-server/lib/db/index.js#L306). To reduce the risk of introducing a bug here I think we could go for an incremental approach and land this first?

Reply:
That sounds good to. I am always good to do thing more incrementally if it limits risk. 👍